app/models/ModelHelpers.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import scala.annotation.tailrec
+
+object ModelHelpers {
```

Comment:
Maybe 

```scala
string.replaceAll("[ ]+", " ").trim
```

Example test
```scala
package models

import org.scalatest.matchers.must.Matchers
import org.scalatest.wordspec.AnyWordSpecLike

class ModelHelpersSpec extends AnyWordSpecLike with Matchers{
  "normaliseSpaces" should {
    "trim an empty string of 1 " in {
      ModelHelpers.normaliseSpaces(" ") mustBe ""
    }

    "leave single characters alone" in {
      ModelHelpers.normaliseSpaces("a b") mustBe "a b"
    }

    "normalise double spacing" in {
      ModelHelpers.normaliseSpaces("a  b") mustBe "a b"
    }

    "normalise triple spacing" in {
      ModelHelpers.normaliseSpaces("a   b") mustBe "a b"
    }

    "normalise quadruple spacing" in {
      ModelHelpers.normaliseSpaces("a    b") mustBe "a b"
    }
  }

  "normaliseSpaces2" should {
    "trim an empty string of 1 " in {
      ModelHelpers.normaliseSpaces2(" ") mustBe ""
    }

    "leave single characters alone" in {
      ModelHelpers.normaliseSpaces2("a b") mustBe "a b"
    }

    "normalise double spacing" in {
      ModelHelpers.normaliseSpaces2("a  b") mustBe "a b"
    }

    "normalise triple spacing" in {
      ModelHelpers.normaliseSpaces2("a   b") mustBe "a b"
    }

    "normalise quadruple spacing" in {
      ModelHelpers.normaliseSpaces2("a    b") mustBe "a b"
    }
  }

}
```

Code 

```scala
package models

import scala.annotation.tailrec

object ModelHelpers {

  def normaliseSpaces(string: String): String = {
    string.replaceAll("[ ]+", " ").trim
  }

  def normaliseSpaces2(string: String): String = {

    @tailrec
    def removeDoubleSpaces(string: String): String = {
      if (!string.contains("  ")) {
        string
      } else {
        removeDoubleSpaces(string.replaceAll("[ ]{2}", " "))
      }
    }

    removeDoubleSpaces(string.trim)
  }



  def normaliseSpaces(string: Option[String]): Option[String] = string.map(normaliseSpaces)

}

```

I think it achieves the same

Reply:
Adopted